synapseclient/core/cache.py

```diff
@@ -87,7 +87,7 @@ def __setattr__(self, key, value):
             value = os.path.expandvars(os.path.expanduser(value))
             # create the cache_root_dir if it does not already exist
             if not os.path.exists(value):
-                os.makedirs(value)
+                os.makedirs(value, exist_ok=True)
```

Comment:
I saw some inconsistent messages popping up in some tests. So leaving it like this as this is valid in python 3.7+